services/demarchesSimplifiees.js

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable camelcase */
+const graphql = require('../utils/graphql');
+const uuid = require('../utils/uuid');
+const date = require('../utils/date');
+const config = require('../utils/config');
+const { DOSSIER_STATE } = require('../utils/dossierState');
+const { default: { getAdeliInfo } } = require('../utils/adeliAPI');
+const { default: { areSimilar } } = require('../utils/string');
+const {
+  default: {
+    getChampsFieldFromId,
+    getChampsIdFromField,
+    getAnnotationsIdFromField,
+  },
+} = require('./champsAndAnnotations');
+
+/**
+ * transform string to boolean
+ * @param {*} inputString 'true' or 'false'
+ */
+function parseTeleconsultation(inputString) {
+  return inputString === 'true';
+}
+
+/**
+ * transform string "speciality1, speciality2" to array ["speciality1", "speciality2"]
+ * as a JSON to store it inside PG
+ */
+function parseTraining(inputString) {
+  if (inputString.includes(',')) {
+    return JSON.stringify(inputString.split(', '));
+  }
+  return JSON.stringify([inputString]);
+}
+
+function getUuidDossierNumber(number) {
+  return uuid.generateUuidFromString(`${config.demarchesSimplifieesId}-${number}`);
+}
+
+function parseDossierMetadata(dossier) {
+  const {
+    state, archived, demandeur, usager, number, groupeInstructeur, champs,
+  } = dossier;
+  const psy = { state, archived };
+
+  psy.dossierNumber = getUuidDossierNumber(number);
+
+  psy.lastName = demandeur.nom.trim();
+  psy.firstNames = demandeur.prenom.trim();
+
+  psy.personalEmail = usager.email.trim();
+
+  psy.region = groupeInstructeur.label;
+
+  const champsToMap = [
+    'departement',
+    'diploma',
+    'adeli',
+    'phone',
+    'address',
+    'email',
+    'languages',
+    'website',
+    'description',
+    'teleconsultation',
+    'training',
+  ];
+  champs.forEach((champ) => {
+    const field = getChampsFieldFromId(champ.id);
+    if (champsToMap.includes(field)) {
```

Comment:
Why do we need this if ?

Reply:
To avoid mapping fields that doesn't exist in the data model (e.g. 'diplomaYear').
This is the bug that was detected by the new integration test (on import DS) :)